src/features/defi/providers/cosmos/components/CosmosManager/Overview/CosmosOverview.tsx

```diff
@@ -0,0 +1,185 @@
+import { ArrowDownIcon, ArrowUpIcon } from '@chakra-ui/icons'
+import { Center } from '@chakra-ui/react'
+import { toAssetId } from '@shapeshiftoss/caip'
+import { DefiModalContent } from 'features/defi/components/DefiModal/DefiModalContent'
+import { Overview } from 'features/defi/components/Overview/Overview'
+import {
+  DefiAction,
+  DefiParams,
+  DefiQueryParams,
+} from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import {
+  isCosmosAssetId,
+  isOsmosisAssetId,
+} from 'plugins/cosmos/components/modals/Staking/StakingCommon'
+import qs from 'qs'
+import { useMemo } from 'react'
+import { FaGift } from 'react-icons/fa'
+import { useTranslate } from 'react-polyglot'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import {
+  MergedActiveStakingOpportunity,
+  useCosmosSdkStakingBalances,
+} from 'pages/Defi/hooks/useCosmosSdkStakingBalances'
+import { useGetAssetDescriptionQuery } from 'state/slices/assetsSlice/assetsSlice'
+import {
+  selectAssetById,
+  selectFirstAccountSpecifierByChainId,
+  selectMarketDataById,
+  selectSelectedLocale,
+  selectTotalBondingsBalanceByAssetId,
+  selectValidatorByAddress,
+} from 'state/slices/selectors'
+import {
+  SHAPESHIFT_COSMOS_VALIDATOR_ADDRESS,
+  SHAPESHIFT_OSMOSIS_VALIDATOR_ADDRESS,
+} from 'state/slices/validatorDataSlice/constants'
+import { useAppSelector } from 'state/store'
+
+import { CosmosEmpty } from './CosmosEmpty'
+import { WithdrawCard } from './WithdrawCard'
+
+export const CosmosOverview = () => {
+  const translate = useTranslate()
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress, assetReference } = query
+
+  const assetNamespace = 'slip44'
+  const stakingAssetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+
+  const opportunities = useCosmosSdkStakingBalances({ assetId: stakingAssetId })
+
+  const opportunity = useMemo(
+    () =>
+      opportunities?.cosmosSdkStakingOpportunities?.find(
+        opportunity => opportunity.address === contractAddress,
+      ) ?? {},
+    [opportunities, contractAddress],
+  ) as MergedActiveStakingOpportunity
```

Comment:
can we avoid this cast too?

Reply:
Removed in df279fe77367c21b3ee7d017acbc8a38a7bebab6